appinventor/components/src/com/google/appinventor/components/runtime/TextBoxBase.java

```diff
@@ -93,6 +102,9 @@ public TextBoxBase(ComponentContainer container, EditText textview) {
       EclairUtil.disableSuggestions(textview);
     }
 
+    // Listen to text changes
+    view.addTextChangedListener(this);
```

Comment:
You may want to consider moving this until after `Text("")` is called later in the constructor.

Reply:
Updated.